### PR TITLE
NN-2893: Add ednpoint which returns cell data for id

### PR DESCRIPTION
**Pull request by Nimphal**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java [hunk 1]
```diff
@@ -260,12 +260,22 @@ private boolean isBlankAddress(final PrisonContactDetail pcd) {
...
+                .map(livingUnit -> transform(livingUnit, true))
                 .filter(Objects::nonNull)
                 .filter(cell -> attribute == null || cell.getAttributes().stream().map(OffenderCellAttribute::getCode).collect(toList()).contains(attribute))
                 .collect(toList());
     }
 
+    @Override
+    public OffenderCell getCellAttributes(@NotNull final Long locationId) {
+        final var livingUnit = livingUnitRepository.findOneByLivingUnitId(locationId);
+        final var offenderCell = livingUnit.map(unit -> transform(unit, false)).orElse(null);
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java:272]: You might be able to do a `orElseThrow (new  EntityNotFoundException.withMessage(String.format("No cell details found for location id %s", locationId)))`

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java [hunk 2]
```diff
@@ -281,9 +291,9 @@ public AgencyEstablishmentTypes getEstablishmentTypes(final String agencyId) {
                 .build();
     }
 
-    private OffenderCell transform(LivingUnit livingUnit) {
+    private OffenderCell transform(LivingUnit livingUnit, Boolean checkCapacity) {
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java:294]: Finals 😨

**Review by JamieDraperUK — APPROVED** [review]
lgtm

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java [hunk 3]
```diff
@@ -260,12 +260,22 @@ private boolean isBlankAddress(final PrisonContactDetail pcd) {
...
         final var livingUnits = livingUnitRepository.findAllByAgencyLocationId(agencyId);
         return livingUnits.stream()
-                .map(this::transform)
+                .map(livingUnit -> transform(livingUnit, true))
                 .filter(Objects::nonNull)
                 .filter(cell -> attribute == null || cell.getAttributes().stream().map(OffenderCellAttribute::getCode).collect(toList()).contains(attribute))
                 .collect(toList());
     }
 
+    @Override
```
Comment (JamieDraperUK) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java:269]: Is it worth adding a service unit test to cover this stuff to complement your integration tests?

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java [hunk 4]
```diff
@@ -281,9 +291,9 @@ public AgencyEstablishmentTypes getEstablishmentTypes(final String agencyId) {
                 .build();
     }
 
-    private OffenderCell transform(LivingUnit livingUnit) {
+    private OffenderCell transform(LivingUnit livingUnit, Boolean checkCapacity) {
         final var agencyLocation = agencyInternalLocationRepository.findOneByDescription(livingUnit.getDescription());
-        if (agencyLocation.isPresent() && agencyLocation.get().isActiveCellWithSpace()) {
+        if (agencyLocation.isPresent() && (!checkCapacity || agencyLocation.get().isActiveCellWithSpace())) {
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java:296]: might be better to use a primitive if you want `checkCapacity` to never be null

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java (hunk 1 above)
Comment (Nimphal) [reply src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java:272]: There's another null possible as a result of the transform, so didn't work neatly with the above.

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java (hunk 2 above)
Comment (Nimphal) [reply src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java]: Of course! Fixed.

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java (hunk 3 above)
Comment (Nimphal) [reply src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java:269]: Yes, it started out way simpler, have added some now.

File: src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java (hunk 4 above)
Comment (Nimphal) [reply src/main/java/uk/gov/justice/hmpps/prison/service/AgencyServiceImpl.java:296]: Thank you, I've done that :)

**Review by JamieDraperUK — APPROVED** [review]
LGTM

**Pull request merged**